src/plugins/instrument_xhr.js

```diff
@@ -293,9 +295,11 @@ class InstrumentXHR {
             const headersCarrier = {};
             tracer.inject(span.context(), opentracing.FORMAT_HTTP_HEADERS, headersCarrier);
             const keys = Object.keys(headersCarrier);
-            keys.forEach((key) => {
-                proxied.setRequestHeader.call(this, key, headersCarrier[key]);
-            });
+            if (self._shouldAddHeadersToRequest(tracer, this.__tracer_url)) {
```

Comment:
Same as the previous comment:

> Is it possible to move this up so that we don't have to create `headersCarrier` or make the call to `tracer.inject`?